### ISSUE TITLE
Browser Extension Changes 20190909 - Phase 2

### ISSUE THREAD

**Issue by rkzack** [ournews/extensions#16]
@webcolor-in this issue will track Phase 2 of the browser extension changes.

**Comment by rkzack** [comment]
![64703428-66f5e380-d47a-11e9-9aed-9f26bd01e57a](https://user-images.githubusercontent.com/35204816/65877938-dca6ee00-e35a-11e9-93bb-99a3cbd2dc41.png)

**Comment by rkzack** [comment]
@webcolor-in to answer your recent questions:

- Where does this "Verified" link take to? 

To the publisher profile, which is constructed based on the `pid` of the publisher. For example, if the `pid` is 123, then the profile URL is `https://our.news/publisher/?pid=123`

- Author tab has repetition of headquarter and verification. 

Instead of grabbing the `results.meta.author` field,  grab the `results.meta.authors` array and use the first entry, which includes `aid`, `name`, `location`, `verified`.   If `verified==1` then display Verified, else display Unverified. 

- So for example: 4 - Accurate (500 Votes) I am using (4 - answer.result. Accurate - answer.label 500 - answer.total)

Yes

- In order to POST answer what key should I use? I assume value should be id of that choice.

-POST the answers as `quicknid` with the value being the value provided in the answers array (should be a number like 1010, or 1011, etc)
-When there are no answers left, display “Thank you for your feedback!” In the space

**Comment by webcolor-in** [comment]
![Annotation 2019-09-30 182905](https://user-images.githubusercontent.com/55145799/65880943-3b4a8700-e3b0-11e9-8031-f0f299f17015.png)

On the last point I am sending `quicknid` but API is throwing this error. I guess it needs location as well so what is the right key?

**Comment by rkzack** [comment]
AH yes, sorry you need to post `quicknid` with the `nid` of the article set as the value, and also include the variable `choicetype` set to the `id` of the selected choice.

**Comment by webcolor-in** [comment]
So `quicknid` and `choicetype` will have same value?

**Comment by rkzack** [comment]
No. `quicknid` is the `nid` of the article.  `choicetype` is the id of the choice the user selected.

**Comment by rkzack** [comment]
Yes, that's expected-- in this case if the user is not logged in, it should fall back to the behavior where it pops up the box asking the user to login or register (from Phase 1 changes).

**Comment by webcolor-in** [comment]
Since the `answer` part is changed and its under `questions`, how do I identify if current answer is already selected one so it can be in bold?

**Comment by rkzack** [comment]
The answered question `id` is returned in the `exturlprivatenid` value `result.mine.answers` array, which contains the `id` of the `choice` that the user has already answered, to be bolded.

**Comment by webcolor-in** [comment]
Thanks, have pushed all my changes in remote.

**Comment by webcolor-in** [comment]
![Annotation 2019-09-30 212511](https://user-images.githubusercontent.com/55145799/65895262-ce8fb680-e3c8-11e9-80b8-85895b0ba2af.png)
Here is how it will look like.

**Comment by rkzack** [comment]
Ok thanks, I will check it out when I return to the office in about 2-3 hours.

**Comment by rkzack** [comment]
@webcolor-in thank you, this all looks very good!   several minor issues found:

- [ ]  On the Summary card. The values under User Ratings should never show anything behind the decimal point, so for example in this screenshot, it should just say "Real News [99%]"

<img width="392" alt="Screen Shot 2019-09-30 at 2 47 42 PM" src="https://user-images.githubusercontent.com/35204816/65906890-4b06a300-e391-11e9-9bef-8090c934a34d.png">

- [ ]  On the Summary card, the results for the Fact-Checks should be links that open into new browser windows, just like they are on the Fact-Check card

<img width="395" alt="2019-09-30_1450" src="https://user-images.githubusercontent.com/35204816/65907144-cc5e3580-e391-11e9-9645-5f4831d0fc11.png">

- [ ] On the Summary card, the results Spin are incomplete.  For example, on this article:

https://our.news/2019/09/01/nasa-admits-that-climate-change-occurs-because-of-changes-in-earths-solar-orbit-and-not-because-of-suvs-and-fossil-fuels/

Spin should be displayed as "Right [74%]"  -- these values come from `results.ratings.spin` (ie. Right) and then `results.ratings.spinvalue` (ie. 74%)

- [ ] On the Summary card, the AI Ratings don't appear to be pulling in the correct data. See below, on this article - `https://www.washingtontimes.com/news/2019/sep/25/revisiting-googles-dominance/?utm_source=RSS_Feed&utm_medium=RSS`-  it should have shown as:

Opinion:   Likely [66%]
News:  Likely [61%]  

Look for the `results.ratings.ai` array, and then show the top two highest values in that array as percentages.

- [ ] On the Summary card, the Results for the questions aren't showing data in the correct position. It should display them as follows (using `https://www.foxnews.com/us/ex-disneyland-princess-convicted-of-being-accessory-in-double-slaying` as an example)

For example, for Trust: 

Results: High [88%] Trust [8 Ratings] 

Results: `label` [`total`]  `result` [`count` Ratings]

Same for the other ones too. 

- [ ]  On the Newstrition card, remove the white "Powered by Our.News" block w/ square logo -- this is no longer needed since the bottom always shows the purple bar with the Our.News logo 

<img width="393" alt="2019-09-30_1448" src="https://user-images.githubusercontent.com/35204816/65907031-8bfeb780-e391-11e9-892f-2db116b2a80f.png">

**Comment by rkzack** [comment]
- [ ] Also -- if an article is not yet indexed, the Summary card is currently showing but with blank data. Instead, it should show the "URL not yet indexed" screen with the Add article button.

**Comment by webcolor-in** [comment]
![Annotation 2019-10-01 080710](https://user-images.githubusercontent.com/55145799/65930189-7e454280-e422-11e9-83f5-cfec445e0711.png)
How do I get to know what be this label? There is 'Certain' 'Unlikely' and this one.

**Comment by webcolor-in** [comment]
> @webcolor-in thank you, this all looks very good! several minor issues found:
> 
> * [x]   On the Summary card. The values under User Ratings should never show anything behind the decimal point, so for example in this screenshot, it should just say "Real News [99%]"
> 
> <img alt="Screen Shot 2019-09-30 at 2 47 42 PM" width="392" src="https://user-images.githubusercontent.com/35204816/65906890-4b06a300-e391-11e9-9bef-8090c934a34d.png">
> 
> * [x]   On the Summary card, the results for the Fact-Checks should be links that open into new browser windows, just like they are on the Fact-Check card
> 
> <img alt="2019-09-30_1450" width="395" src="https://user-images.githubusercontent.com/35204816/65907144-cc5e3580-e391-11e9-9645-5f4831d0fc11.png">
> 
> * [x]  On the Summary card, the results Spin are incomplete.  For example, on this article:
> 
> https://our.news/2019/09/01/nasa-admits-that-climate-change-occurs-because-of-changes-in-earths-solar-orbit-and-not-because-of-suvs-and-fossil-fuels/
> 
> Spin should be displayed as "Right [74%]" -- these values come from `results.ratings.spin` (ie. Right) and then `results.ratings.spinvalue` (ie. 74%)
> 
> * [x]  On the Summary card, the AI Ratings don't appear to be pulling in the correct data. See below, on this article - `https://www.washingtontimes.com/news/2019/sep/25/revisiting-googles-dominance/?utm_source=RSS_Feed&utm_medium=RSS`-  it should have shown as:
> 
> Opinion: Likely [66%]
> News: Likely [61%]
> 
> Look for the `results.ratings.ai` array, and then show the top two highest values in that array as percentages.
> 
> * [x]  On the Summary card, the Results for the questions aren't showing data in the correct position. It should display them as follows (using `https://www.foxnews.com/us/ex-disneyland-princess-convicted-of-being-accessory-in-double-slaying` as an example)
> 
> For example, for Trust:
> 
> Results: High [88%] Trust [8 Ratings]
> 
> Results: `label` [`total`] `result` [`count` Ratings]
> 
> Same for the other ones too.
> 
> * [x]   On the Newstrition card, remove the white "Powered by Our.News" block w/ square logo -- this is no longer needed since the bottom always shows the purple bar with the Our.News logo
> 
> <img alt="2019-09-30_1448" width="393" src="https://user-images.githubusercontent.com/35204816/65907031-8bfeb780-e391-11e9-892f-2db116b2a80f.png">

All these done

**Comment by webcolor-in** [comment]
All above tweaks / fixes done and code changes pushed to remote.

**Comment by rkzack** [comment]
Hi @webcolor-in ok great progress, thank you. A few final items:

- [ ] Re your last question: "How do I get to know what be this label? There is 'Certain' 'Unlikely' and this one." --  get them from the `results.ratings.ai_labels` array

- [ ] On the Summary card -- see screenshot-- could you please right-align the values (not the labels) to make it look like the rest of the nutrition label?  

<img width="387" alt="2019-10-01_0916" src="https://user-images.githubusercontent.com/35204816/65968812-bdc75b00-e431-11e9-8fe4-36a7ff07927d.png">


- [ ] After adding a new article to the Index, it's currently taking you to the Fact-Check card. Instead, after adding a new article, can it open the Summary card?


- [ ] We want to change the behavior for if an article isn't yet indexed to the following:  1)  Open on the Newstrition card.  2)  In the white box above the tabs, put the message "This article has not yet been indexed in our database. Would you like to add it?"  along with a small button or text link to "Add Article";  3)  After the article is indexed, open the Summary tab.


- [ ] In the questions, there seems to be a problem with the Results, as they are not matching the API results. See attached screenshot. We checked API results and they are correct but they aren't showing up properly here. So this one may be a bug. The article we're testing with is: `https://www.foxnews.com/us/ex-disneyland-princess-convicted-of-being-accessory-in-double-slaying`

<img width="391" alt="2019-10-01_0950" src="https://user-images.githubusercontent.com/35204816/65968781-aee0a880-e431-11e9-83da-e37e520d6d48.png">

**Comment by webcolor-in** [comment]
For `Results` I am doing following as you mentioned before:

Results: High [88%] Trust [8 Ratings]
Results: label [total] result [count Ratings]

![view](https://user-images.githubusercontent.com/55145799/65972743-e9753000-e487-11e9-869c-79b8ab12b39c.png)
![api](https://user-images.githubusercontent.com/55145799/65972757-ed08b700-e487-11e9-9506-6bb34222993e.png)

**Comment by rkzack** [comment]
Re the Results issue-- ok we will check the API, maybe something wrong there.

Re the 2-3 Hours to change article index behavior-- please proceed asap.

**Comment by webcolor-in** [comment]
Except that rest tweak is complete and code pushed to remote.

**Comment by webcolor-in** [comment]
done

**Comment by rkzack** [comment]
Ok great, one more small issue:

- Summary card - see attached screenshot-- please make sure these values don't wrap -- this is on this article:  `http://actionnews3.com/rat-meat-has-been-sold-as-chicken-wings/`

<img width="387" alt="2019-10-01_1410" src="https://user-images.githubusercontent.com/35204816/65988383-b44fea00-e455-11e9-8d93-7fc57e3888b2.png">

**Comment by webcolor-in** [comment]
I increased column size for fact-checks values so it shouldn't wrap unless text is too long to fit. Code pushed.
